### PR TITLE
Frontend and backend improvements 

### PR DESCRIPTION
**Pull request by aleksejlopasov**
Most of HTML elements or their parts are now rendered from templates. The code is shorter and readable.
'None' option has been added to image versioning to keep image version/URI in YAML file.
Cluster properties are now supported.
Primary/Secondary worker sections hide in UI when public example-single-node.yaml is selected.

### CODE REVIEW

File: dataprocspawner/spawner.py [hunk 1]
```diff
@@ -1218,14 +1218,32 @@ def _apply_users_configs(self, cluster_data):
...
+        self.log.info(f'Failed to process Dataproc cluster properties:\n\t{e}')
+
     if self.user_options.get('image_version'):
       if self.user_options.get('custom_image'):
         config['master_config']['image_uri'] = self.user_options.get('custom_image')
         config['worker_config']['image_uri'] = self.user_options.get('custom_image')
         config['software_config']['image_version'] = \
-          self._get_custom_image_version(self.user_options.get('custom_image'))
+          self._get_image_version(self.user_options.get('custom_image'))
       else:
```
Comment (m-mayran) [inline dataprocspawner/spawner.py:1239]: Could you add tests for this?

File: dataprocspawner/spawner.py [hunk 2]
```diff
@@ -1218,14 +1218,32 @@ def _apply_users_configs(self, cluster_data):
...
+        config['master_config'].pop('image_uri', None)
+        config['worker_config'].pop('image_uri', None)
         config['software_config']['image_version'] = self.user_options.get('image_version')
+    # else:
```
Comment (m-mayran) [inline dataprocspawner/spawner.py:1243]: Remove?

**Review by m-mayran — COMMENTED** [review]
Added a couple of small things. 

Could you add some tests to check image URIs and image version when user decides to overwrite? I think they are missing (even if the code works from my tries)

For example:
- image_version should adapt to the one provided in the custom image
- image version should not conflict with image_uris already existing in YAML.

File: dataprocspawner/spawner.py (hunk 2 above)
Comment (aleksejlopasov) [reply dataprocspawner/spawner.py]: My loss =)
Done

**Comment by aleksejlopasov** [comment]
> Added a couple of small things.
> 
> Could you add some tests to check image URIs and image version when user decides to overwrite? I think they are missing (even if the code works from my tries)
> 
> For example:
> 
> * image_version should adapt to the one provided in the custom image
> * image version should not conflict with image_uris already existing in YAML.

Please check the recently added tests (the last two) for checking user-defined image_version/custom_image. If that is what you expect?

File: dataprocspawner/customize_cluster.py [hunk 3]
```diff
@@ -29,52 +29,29 @@ def get_base_cluster_html_form(configs, locations_list, jhub_region):
...
-                Cluster's configuration
-              </span>
-            </span>
-            <span class="mdc-notched-outline__trailing"></span>
-          </span>
-        </div>
-      <div class="mdc-select__menu mdc-menu mdc-menu-surface">
-        <ul class="mdc-list">"""
+      <div class="mdc-select mdc-select-decision jupyter-select mdc-select--outlined"
+        data-decision-value="gs://dataproc-spawner-dist/example-configs/example-single-node.yaml"
```
[the remark this answers is not in the document]
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:33]: That means we have to check the file content every time the selector changes in the UI. That requires additional JS functionality + additional backend to process YAML files. I will discuss it with UI team.

[the remark this answers is not in the document]
Comment (m-mayran) [reply dataprocspawner/customize_cluster.py:33]: Agreed. That's why I am saying "idea to debate". Might not be worth it. Not for this PR anyway.

**Review by m-mayran — APPROVED** [review]
LGTM

**Pull request merged**